src/handlers.js

```diff
@@ -120,9 +120,14 @@ Handlers.prototype.completed = function(message) {
 
   // Get task definition
   return this.queue.task(message.payload.status.taskId).then(function(task) {
-    // Create default expiration date
-    var expires = new Date();
-    expires.setDate(expires.getDate() + 365);
+    
+    // Create expiration date
+    var expires = new Date();    
+    if(typeof(task.expires) !== 'undefined') {
+      expires.setDate(task.expires);
+    } else {
+      expires.setDate(expires.getDate() + 365);
```

Comment:
Ok

2016-04-27 8:22 GMT+02:00 Jonas Finnemann Jensen <notifications@github.com>:

> In src/handlers.js
> <https://github.com/taskcluster/taskcluster-index/pull/17#discussion_r61209321>
> :
>
> > @@ -120,9 +120,14 @@ Handlers.prototype.completed = function(message) {
> >
> >    // Get task definition
> >    return this.queue.task(message.payload.status.taskId).then(function(task) {
> > -    // Create default expiration date
> > -    var expires = new Date();
> > -    expires.setDate(expires.getDate() + 365);
> > +
> > +    // Create expiration date
> > +    var expires = new Date();
> > +    if(typeof(task.expires) !== 'undefined') {
> > +      expires.setDate(task.expires);
> > +    } else {
> > +      expires.setDate(expires.getDate() + 365);
>
> Let's default this to task.created + 1 year... That's what we use
> elsewhere... Sorry forgot to mention that earlier...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/taskcluster/taskcluster-index/pull/17/files/d8cb03bf50bf4a22754b024e3cb093e300fdba8d#r61209321>
>